### ISSUE TITLE
Unexpected results in VB after a class Refactor Rename

### ISSUE THREAD

**Issue by dgrunwald** [icsharpcode/SharpDevelop#197]
[SD-1795](http://bugtracker.sharpdevelop.net/issue/ViewIssue.aspx?id=1795), originally created on 12/9/2010 18:16:54 by Eusebiu Marcu

[http://community.sharpdevelop.net/forums/t/12379.aspx](http://community.sharpdevelop.net/forums/t/12379.aspx)

---

**Comment from Siegfried Pammer on 12/9/2010 18:24:32**:
This is a general bug in the rename refactoring. It occurs with C# too.
Might be because Rename uses normal text search and replace.

---

**Comment from Daniel Grunwald on 12/9/2010 18:37:50**:
This is a design limitation in the NRefactoryResolver.
 It is caused by the C# (and VB) rule that allows stuff like "public
Color Color { get; set; }".
 Basically the name "testName" can refer to both the type and to the
field, depending on how it's used.
 I think we won't be able to fix this in the SD.Dom implementation, this
issue will have to wait for the new NRefactory.

---

**Comment from Eusebiu Marcu on 8/11/2011 22:00:17**:
Also consider these:
1. Renaming properties doesn't rename when it is used (only the
definition)
2. Renaming indexers (defined
as Public Property MyProp(ByVal i As Integer, ByVal j As Integer) As String) doesn't
rename anything - looks like the renaming is just replacing the text.
